### PR TITLE
wpcomsh: Fix PHP 8 fatal on managed plugins

### PR DESCRIPTION
**Pull request by tbradsha**
In some obscure scenarios, a plugin update check would trigger a PHP fatal on PHP 8. Reported in Automattic/jpop-issues#9074.

The fix is simple: cast the `get_option()` return value to an int.

I checked for other instances of math operations next to `get_option()` and fixed the three others I found.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

While the fix is simple, I struggled to come up with a way to reproduce this consistently. I was able to confirm the error on one of my test sites after poking at it for some time, but the trigger is unclear. What I do know is:

* One needs to have `XMLRPC_REQUEST` defined.
* One needs to have a `auto_updater.lock` option set to a non-numeric value (e.g. an empty string).
* The plugin update mechanism needs to trigger.

For the latter point, though, there are all sorts of preventative conditions in core (and probably elsewhere), so I don't have testing steps.

Instead, just look at the code and you'll see what it does.

### CODE REVIEW

File: projects/plugins/jetpack/sal/class.json-api-date.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public static function format_date( $date_gmt, $date = null ) {
 						$offset = $timezone->getOffset( $date_time );
 					}
 				} else {
-					$offset = 3600 * get_option( 'gmt_offset' );
+					$offset = 3600 * (int) get_option( 'gmt_offset' );
```
Comment (anomiex) [inline projects/plugins/jetpack/sal/class.json-api-date.php]: I wonder if this one might be a float.

I know there are some timezones that are a non-integer number of hours offset from UTC, but I don't know whether this 'gmt_offset' option handles that or not.

**Review by anomiex — COMMENTED** [review]
Seems like this will mostly take care of the errors. I wonder if it's just hiding a problem that caused the option to be a non-numeric string in the first place though.

File: projects/plugins/jetpack/sal/class.json-api-date.php (hunk 1 above)
Comment (tbradsha) [reply projects/plugins/jetpack/sal/class.json-api-date.php]: Good thought; reverted in a159a56dff.

Seems the same thing happened in WP core:
WordPress/WordPress@6e83ac559549a3b1a391286ef57e8680f97d498d
WordPress/WordPress@b1d826fb6e0bd5f8ffcad511658e8de8a60e6a42

**Comment by tbradsha** [comment]
> I wonder if it's just hiding a problem that caused the option to be a non-numeric string in the first place though.

That's why I spent a bit of time trying to reproduce, but I didn't have much success triggering it on most sites. The option is created by `WP_Upgrader::create_lock()` (which expects an optional `int`), so if we're creating it wrong via our plugins Phan should catch it (I just tested and it does, though `PhanTypeMismatchArgumentProbablyReal` is currently suppressed in the one file that calls said method).

In other words, it's likely something outside of Jetpack is manipulating things unexpectedly, and the best we can do is handle it gracefully.

**Review by anomiex — DISMISSED** [review]
🤷

**Review by anomiex — APPROVED** [review]
